src/components/Proposal/Create/PluginForms/CreateKnownGenericPluginProposal.tsx

```diff
@@ -64,23 +67,68 @@ interface IState {
   tags: Array<string>;
 }
 
-class CreateKnownPluginProposal extends React.Component<IProps, IState> {
+const setInitialFormValues = (props: IProps) => {
+
+  const defaultValues: IFormValues = {
+    description: "",
+    title: "",
+    url: "",
+    currentActionId: "",
+    tags: [],
+  };
+
+  const actions = props.genericPluginInfo.actions();
+  const daoAvatarAddress = props.daoAvatarAddress;
+  actions.forEach((action) => action.getFields().forEach((field: ActionField) => {
+    if (typeof (field.defaultValue) !== "undefined") {
+      if (field.defaultValue === "_avatar") {
+        defaultValues[field.name] = daoAvatarAddress;
+      } else {
+        defaultValues[field.name] = field.defaultValue;
+      }
+    } else {
+      switch (field.type) {
+        case "uint64":
+        case "uint256":
+        case "bytes32":
+        case "bytes":
+        case "address":
+        case "string":
+          defaultValues[field.name] = "";
+          break;
+        case "bool":
+          defaultValues[field.name] = 0;
+          break;
+        case "address[]":
+          defaultValues[field.name] = [""];
+          break;
+      }
```

Comment:
Is it not complaining of missing `default` keyword?

Reply:
It is not complaining.  This is old code that I would hesitate to mess with.

Reply:
This line looks suspect to me:

```
defaultValues[field.name] = [""]
```

seems like should more likely be:

```
defaultValues[field.name] = []
```